annotated_test.go

```diff
@@ -176,3 +176,159 @@ func TestAnnotatedString(t *testing.T) {
 		})
 	}
 }
+
+func TestAnnotate(t *testing.T) {
+	type a struct{}
+	type b struct{ a *a }
+	type c struct{ b *b }
+	newA := func() *a { return &a{} }
+	newB := func(a *a) *b {
+		return &b{a}
+	}
+	newC := func(b *b) *c {
+		return &c{b}
+	}
+
+	t.Run("Provide with optional", func(t *testing.T) {
+		app := fxtest.New(t,
+			fx.Provide(
+				fx.Annotate(newB, fx.ParamTags(`name:"a" optional:"true"`)),
+			),
+			fx.Invoke(newC),
+		)
+		defer app.RequireStart().RequireStop()
+		require.NoError(t, app.Err())
```

Comment:
If this is a failure case, we should assert something about the error message.
It's common to say something like the following in a test:

```
assert.Contains(t, err.Error(), "subset of the message")
```